### PR TITLE
feat: update go 1.20

### PR DESCRIPTION
**Pull request by devhaozi**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,8 +19,6 @@ jobs:
       - uses: actions/setup-go@v4
         with:
           go-version: ${{ matrix.go }}
-      - name: Install dependencies 📦
-        run: go mod tidy
```
Comment (hwbrzzl) [inline .github/workflows/test.yml:23]: Q: Why remove this?

**Review by hwbrzzl — CHANGES_REQUESTED** [review]
Based on the official document, I think we can upgrade to 1.20 at most, 1.21 will have more effect, and we need to merge this PR to master.

File: .github/workflows/test.yml (hunk 1 above)
Comment (devhaozi) [reply .github/workflows/test.yml:23]: > Q: Why remove this?

It will throw a error to request update go version.

Comment (hwbrzzl) [reply .github/workflows/test.yml:23]: Is there an error if we upgrade to 1.20?

Comment (devhaozi) [reply .github/workflows/test.yml:23]: > Is there an error if we upgrade to 1.20?

Yes, the error will be:
`go: go.mod file indicates go 1.20, but maximum version supported by tidy is 1.18`

Comment (hwbrzzl) [reply .github/workflows/test.yml:23]: Hence we must remove 1.18 and 1.19 for the test.yml file if we want to upgrade to 1.20. Okay, let's remove 1.18 and 1.19 if we must do it.

Comment (hwbrzzl) [reply .github/workflows/test.yml:23]: Sorry for this, you have a PR to remove 1.18 and 1.19 just now.

Comment (devhaozi) [reply .github/workflows/test.yml:23]: > Sorry for this, you have a PR to remove 1.18 and 1.19 just now.

That PR is closed, If needed, I will finished in this PR.

Comment (hwbrzzl) [reply .github/workflows/test.yml:23]: Sure, let's remove 1.18 and 1.19 and add back `Install dependencies`.

File: .github/workflows/test.yml
Comment (hwbrzzl) [inline .github/workflows/test.yml:1]: We need to remove 1.18 and 1.19(you said there is an error if we don't remove them).

Comment (devhaozi) [reply .github/workflows/test.yml:1]: > We need to remove 1.18 and 1.19(you said there is an error if we don't remove them).

I forgot it, fixed.

**Review by hwbrzzl — APPROVED** [review]
LGTM

**Pull request merged**